### ISSUE TITLE
♿ 🐞 Bug Report: Toggle widget:  "Elements with an ARIA [role] that require children to contain a specific [role] are missing some or all of those required children"

### ISSUE THREAD

**Issue by bethgee** [elementor/elementor#22021]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Getting an error on Google Page Speed which says "Elements with an ARIA [role] that require children to contain a specific [role] are missing some or all of those required children. " - in reference to all toggle widgets.

I worked out it was a theme issue but I replicated it on Hello Elementor, Hello Child theme, GeneratePress and Astra. However, it disappears with Twenty Twenty-Three.

### Steps to reproduce

Google PageSpeed Insights
Enter domain 
Issue listed under Accessibility

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

[system-info-www.carelawyers.com.au-12-04-2023.txt](https://github.com/elementor/elementor/files/11214895/system-info-www.carelawyers.com.au-12-04-2023.txt)

**Comment by gorkemuluser** [comment]
Hi, can you check by trying the below steps?

1. Log in to WordPress and go to the Elementor panel.
2. Select the Hello Elementor theme and open a page.
3. In the Elementor editor, add a Toggle widget.
4. Select the widget and go to the "Advanced" tab.
5. In the "Advanced" tab, add the following code to the "CSS Classes" field:
`elementor-toggle-content { role="region"; }`
6. Save and publish the changes.

This code assigns the "region" role to the content of the Toggle widget, which can help resolve the issue reported in Google PageSpeed Insights.

**Comment by bethgee** [comment]
> Hi, can you check by trying the below steps?
> 
>     1. Log in to WordPress and go to the Elementor panel.
> 
>     2. Select the Hello Elementor theme and open a page.
> 
>     3. In the Elementor editor, add a Toggle widget.
> 
>     4. Select the widget and go to the "Advanced" tab.
> 
>     5. In the "Advanced" tab, add the following code to the "CSS Classes" field:
>        `elementor-toggle-content { role="region"; }`
> 
>     6. Save and publish the changes.
> 
> 
> This code assigns the "region" role to the content of the Toggle widget, which can help resolve the issue reported in Google PageSpeed Insights.


I tried that, it makes no difference at all

**Comment by gorkemuluser** [comment]
Could you please add the following code to the very bottom of the functions.php file:

```
add_filter( 'elementor_pro/utils/get_html_attributes/role', function( $html, $tag ) {
return ( 'elementor-toggle' === $tag ) ? 'role="switch"' : $html;
}, 10, 2 );
```

After adding the code, could you please save the file and check if the issue still persists?

**Comment by tarsonis123** [comment]
The same ARIA issue is also present for the accordion widget. Is the fix implemented there too for the upcoming RC?

**Comment by Betolazaro** [comment]
> Você poderia adicionar o seguinte código bem no final do arquivo functions.php:
> 
> ```
> add_filter( 'elementor_pro/utils/get_html_attributes/role', function( $html, $tag ) {
> return ( 'elementor-toggle' === $tag ) ? 'role="switch"' : $html;
> }, 10, 2 );
> ```
> 
> Depois de adicionar o código, você poderia salvar o arquivo e verificar se o problema ainda persiste?

Funcionou pra alguém? Problema muito grave esse, traz prejuízos pra quem precisa de classificação no google. Meu site desceu 14 posições.

**Comment by yllip** [comment]
This is still not fixed, for a reference please [test this url](https://kaliumtheme.com/)
